hstest/testing/execution/searcher/cpp_searcher.py

```diff
@@ -0,0 +1,25 @@
+import re
+
+from hstest.testing.execution.filtering.main_filter import MainFilter
+from hstest.testing.execution.runnable.runnable_file import RunnableFile
+from hstest.testing.execution.searcher.base_searcher import BaseSearcher
+
+
+class CppSearcher(BaseSearcher):
+
+    @property
+    def extension(self) -> str:
+        return '.cpp'
+
+    def search(self, where: str = None) -> RunnableFile:
+        main_func_searcher = re.compile(r'(^|\n)\s*int\s+main\s*\(\s*\)', re.M)
```

Comment:
I saw `(^|\n) *int +main\\(.*\\)` in the Java PR.
Could you use the same template in both PRs?
I like combined one: `(^|\n)\s*int\s+main\s*\(.*\)` because
1. It has `\s` and it is generally more correct than a space.
2. It has `.*` in inside the `( )` since brackets can contain something like `void` or `int argc, char *argv[]` see [examples](https://mr-kumar-abhishek.github.io/blog/2020/04/11/overview-of-main-function-in-c-and-c-plus-plus).

The 2nd point is crucial, because main methods generally have arguments and not just spaces, therefore the current regexp is slightly incorrect.

Reply:
Many thanks to you, you are right, I missed the presence of arguments